Remove incorrect usages of skipIfTorchDynamo

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #117139
* #117133
* #117129
* #117128
* #117127
* __->__ #117114

Using `@skipifTorchDynamo` is wrong, the correct usage is
`@skipIfTorchDynamo()` or `@skipIfTorchDynamo("msg")`. This would cause
tests to stop existing.
Added an assertion for this and fixed the incorrect callsites.

cc @mcarilli @ptrblck @leslie-fang-intel @jgong5

zou3519 committed:
Remove incorrect usages of skipIfTorchDynamo

Using `@skipifTorchDynamo` is wrong, the correct usage is
`@skipIfTorchDynamo()` or `@skipIfTorchDynamo("msg")`. This would cause
tests to stop existing.
Added an assertion for this and fixed the incorrect callsites.

[ghstack-poisoned]

zou3519 committed:
Update on "Remove incorrect usages of skipIfTorchDynamo"


Using `skipifTorchDynamo` is wrong, the correct usage is
`skipIfTorchDynamo()` or `skipIfTorchDynamo("msg")`. This would cause
tests to stop existing.
Added an assertion for this and fixed the incorrect callsites.

cc mcarilli ptrblck leslie-fang-intel jgong5

[ghstack-poisoned]

zou3519 committed:
Update on "Remove incorrect usages of skipIfTorchDynamo"


Using `skipifTorchDynamo` is wrong, the correct usage is
`skipIfTorchDynamo()` or `skipIfTorchDynamo("msg")`. This would cause
tests to stop existing.
Added an assertion for this and fixed the incorrect callsites.

cc mcarilli ptrblck leslie-fang-intel jgong5

[ghstack-poisoned]

kit1980 commented:
Hm, usually for decorators people expect to be able to omit the parentheses if there are no parameters...

zou3519 committed:
Update on "Fix skipIfTorchDynamo to work when passed nothing"


Previously, code that looked like:
```
skipifTorchDynamo
def test_foo(self):
    pass
```
would delete the test. The correct usage was `skipIfTorchDynamo()`.
This PR fixes that.

[ghstack-poisoned]

zou3519 committed:
Update on "Remove incorrect usages of skipIfTorchDynamo"


Using `skipifTorchDynamo` is wrong, the correct usage is
`skipIfTorchDynamo()` or `skipIfTorchDynamo("msg")`. This would cause
tests to stop existing.
Added an assertion for this and fixed the incorrect callsites.

[ghstack-poisoned]

zou3519 commented:
> Hm, usually for decorators people expect to be able to omit the parentheses if there are no parameters...

It's a bit too late for this. If we allow omitting the parenthesis, then the function needs to be the first arg to the decorator, which would break usages that look like `skipIfTorchDynamo("reason")`. We can rewrite existing usages to instead be skipIfTorchDynamo(msg="reason")` to avoid ambiguity, but that's a larger task -- this goal of this PR is to stem the bleeding.

kit1980 commented:
> If we allow omitting the parenthesis, then the function needs to be the first arg to the decorator, which would break usages that look like `skipIfTorchDynamo("reason")`.

Maybe you can check the type of the first positional argument. If it's a string, then it's msg.

zou3519 committed:
Update on "Remove incorrect usages of skipIfTorchDynamo"


Using `skipifTorchDynamo` is wrong, the correct usage is
`skipIfTorchDynamo()` or `skipIfTorchDynamo("msg")`. This would cause
tests to stop existing.
Added an assertion for this and fixed the incorrect callsites.

[ghstack-poisoned]
